build/build.go

```diff
@@ -716,10 +504,15 @@ func BuildWithResultHandler(ctx context.Context, nodes []builder.Node, opt map[s
 		}
 	}
 
-	m, clients, err := resolveDrivers(ctx, nodes, opt, w)
+	driverRes := newDriverResolver(nodes)
```

Comment:
nit: if the object only has one usable method this call should still be `resolveDrivers()`